config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationPackage.java

```diff
@@ -644,9 +645,15 @@ private void preprocess(File appDir, File dir, Zone zone) throws IOException {
     private void validateServicesFile() throws IOException {
         File servicesFile = getServicesFile();
         if ( ! servicesFile.exists())
-            throw new IllegalArgumentException(SERVICES + " does not exist in application package");
+            throw new IllegalArgumentException(SERVICES + " does not exist in application package. " +
+                                               "There are " + filesInApplicationPackage() + " files in the directory");
         if (IOUtils.readFile(servicesFile).isEmpty())
-            throw new IllegalArgumentException(SERVICES + " in application package is empty");
+            throw new IllegalArgumentException(SERVICES + " in application package is empty. " +
+                                               "There are " + filesInApplicationPackage() + " files in the directory");
+    }
+
+    private long filesInApplicationPackage() {
+        return uncheck(() -> Files.list(appDir.toPath()).count());
```

Comment:
This should be closed
> This method must be used within a try-with-resources statement or similar control structure to ensure that the stream's open directory is closed promptly after the stream's operations have completed.
(https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/nio/file/Files.html#list(java.nio.file.Path))

Reply:
Thanks, done in https://github.com/vespa-engine/vespa/pull/32436